linkerd/stack/src/gate.rs

```diff
@@ -84,27 +100,28 @@ impl Drop for Rx {
 
 impl Tx {
     /// Returns when all associated `Rx` clones are dropped.
-    pub async fn closed(&self) {
-        self.0.closed.notified().await;
+    pub async fn lost(&self) {
+        self.0.closed().await
     }
 
     /// Opens the gate.
     pub fn open(&self) {
-        if !self.0.open.swap(true, std::sync::atomic::Ordering::Release) {
+        if self.0.send(State::Open).is_ok() {
             debug!("Gate opened");
-            self.0.notify.notify_waiters();
+        }
+    }
+
+    /// Opens the gate.
```

Comment:
i don't think this comment is quite right --- `open` is the function that opens the gate.